transport/socks5/stream_dialer.go

```diff
@@ -55,78 +61,110 @@ func (c *streamDialer) DialStream(ctx context.Context, remoteAddr string) (trans
 	}()
 
 	// For protocol details, see https://datatracker.ietf.org/doc/html/rfc1928#section-3
+	// Creating a single buffer for method selection, authentication, and connection request
+	var buffer []byte
```

Comment:
We were pre-allocating before to avoid multiple allocations. Please keep doing the same.